test/lit/passes/optimize-instructions.wast

```diff
@@ -12212,4 +12212,27 @@
       )
     )
   )
+  ;; CHECK:      (func $if-dont-change-to-unreachable (param $x i32) (param $y i32) (param $z i32) (result i32)
+  ;; CHECK-NEXT:  (if (result i32)
+  ;; CHECK-NEXT:   (local.get $x)
+  ;; CHECK-NEXT:   (return
+  ;; CHECK-NEXT:    (local.get $y)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (return
+  ;; CHECK-NEXT:    (local.get $z)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $if-dont-change-to-unreachable (param $x i32) (param $y i32) (param $z i32) (result i32)
```

Comment:
In principle this would be fine for `select`s, right? But we choose not to handle them here since DCE will take care of that. Can we add a test showing that case as well?

Reply:
No, select would work here actually, and we do handle it (no reason not to - it would be more code to avoid). There is already a testcase showing it works for if (when the if has unreachable type), I'll add one for select.

Reply:
Oh, actually it doesn't work for select as we execute both sides, and a return has side effects. So for now it's ignored for that reason. Testcase added.